kitsune/settings.py

```diff
@@ -1209,3 +1209,4 @@ def filter_exceptions(event, hint):
 TOLL_FREE_REGEX = re.compile(r'^.*8(00|33|44|55|66|77|88)[0-9]\d{6,}$')
 SIMPLE_DOMAIN_IP_REGEX = re.compile(r'(?:[^\s/]+\.[^\s0-9/.]{2,})|(?:\d{1,3}\.){3}\d{1,3}')
 REGEX_TIMEOUT = config('REGEX_TIMEOUT', default=5, cast=int)
+VANITY_PHONE_NUMBER_REGEX = re.compile(r'(\+?1?)-?([0-9]{3})-?([a-zA-Z0-9]{7,8})$')
```

Comment:
Do we have an authoritative source for what form vanity numbers will take? Can we add some tests to tests against things that are and things that aren't vanity numbers?

Reply:
From https://en.wikipedia.org/wiki/North_American_Numbering_Plan#Alphabetic_mnemonic_system, I believe the regexp should be `[0-9]{3}-?[a-zA-Z2-9][a-zA-Z0-9]{6}`:
* removed unnecessary capturing groups
* no `(\+?1?)-?` is needed because it's all marked as optional (`?`)
* `[a-zA-Z2-9]` because from what I can see the "first digit in the NXX must be [2-9]" rule still applies
* `{6}` because I can't see any source which allows the xxxx to be longer than 4 digits
* no `$` because it could appear anywhere in a line of text

Also, this will match all vanity numbers, *and* all non-vanity ones in the NANP - should we just rename the regex to something like `NANP_REGEX`?